src/components/form-elements/Input.jsx

```diff
@@ -23,21 +23,35 @@ export const TYPE = {
   TIME: 'time',
   URL: 'url',
   WEEK: 'week'
-};
+});
 
-export const SIZE = {
+export const SIZE = Object.freeze({
   SMALL: 'small',
   LARGE: 'large',
   NORMAL: 'normal'
-};
+});
 
-export const COLOR = {
+export const COLOR = Object.freeze({
   NORMAL: 'normal',
   LIGHT: 'light',
   LIGHT_ALT: 'light-alt'
+});
+
+type PropsType = {
+  type?: $Values<typeof TYPE>,
+  value?: string | number,
+  size?: $Values<typeof SIZE>,
+  color?: $Values<typeof COLOR>,
+  valid?: boolean,
+  invalid?: boolean,
+  fullWidth?: boolean,
+  noBorder?: boolean,
+  withIcon?: boolean,
+  className?: string,
+  setInputRef?: (ref: ?HTMLElement) => mixed
 };
 
-const Input = props => {
+const Input = (props: PropsType) => {
```

Comment:
If we don't pass ref to input anywhere right now maybe we could use ref forwarding https://reactjs.org/docs/forwarding-refs.html#forwarding-refs-to-dom-components